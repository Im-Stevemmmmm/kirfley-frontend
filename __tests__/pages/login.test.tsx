import { gql } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import { act, render } from '@testing-library/react';
import ReactDOM from 'react-dom';
import Login from '../../pages/login';
import { GQLMocks } from '../__utils__/apollo';

describe('login form', () => {
  it('renders properly', () => {
    const container = document.createElement('div');

    act(() => {
      ReactDOM.render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Login />
        </MockedProvider>,
        container
      );
    });

    expect(container).toMatchSnapshot();
  });

  describe('fields', () => {
    beforeEach(() => {
      render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Login />
        </MockedProvider>
      );
    });
  });
});

const AUTHENTICATE_USER = gql`
  query AuthenticateUser($email: String!, $password: String!) {
    login(options: { email: $email, password: $password }) {
      successful
    }
  }
`;

enum FieldValues {
  EMAIL = 'example@email.com',
  PASSWORD = '1234',
}

const mocks: GQLMocks[] = [
  {
    request: {
      query: AUTHENTICATE_USER,
      variables: {
        email: FieldValues.EMAIL,
        password: FieldValues.PASSWORD,
      },
    },
    result: {
      data: {
        login: {
          successful: true,
        },
      },
    },
  },
];
