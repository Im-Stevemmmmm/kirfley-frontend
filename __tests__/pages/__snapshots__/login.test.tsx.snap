// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login form renders properly 1`] = `
<div>
  <div
    id="root"
  >
    <div
      id="leftColumn"
    >
      <form>
        <h1>
          Log In
        </h1>
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          type="email"
          value=""
        />
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
        <button
          type="submit"
        >
          Log In
        </button>
      </form>
      <div
        id="background"
      />
    </div>
    <div
      id="rightColumn"
    />
  </div>
</div>
`;
